lib/rules/lint-no-unnecessary-component-helper.js

```diff
@@ -7,13 +7,29 @@ const ERROR_MESSAGE = 'Invoke component directly instead of using `component` he
 
 module.exports = class NoUnnecessaryComponentHelper extends Rule {
   visitor() {
-    function checkNode(node) {
-      if (
+    function isComponentHelper(node) {
+      return (
+        node &&
```

Comment:
These null checks (`node && node.path`) should be removed. If someone passes null to this function, it's a problem and we should fail fast. Instead, you can use type checks from AstNodeInfo on node if needed.

Reply:
I added them in 1b5892ea738a726ccf08542dff4f2f68aac1c7dd, because static attributes don't have a `path`, causing a crash in L14, since `AstNodeInfo.isPathExpression(node.path)` does not handle nullish values.

https://github.com/ember-template-lint/ember-template-lint/blob/bc849e2376990ca11ec71396fa90f8e449b028f3/lib/helpers/ast-node-info.js#L31-L33

Reply:
I can either:

- keep everything as it is
- add `node &&` to the `isPathExpression` (and other helpers?)
- move the `node && node.path` out of `isComponentHelper` and into the `filter` arrow function in [`recordAngleBracketArguments`](https://github.com/buschtoens/ember-template-lint/blob/no-unnecessary-component-helper-with-angle-brackets/lib/rules/lint-no-unnecessary-component-helper.js#L23)